### DISCUSSION TITLE
[GATech]Fast MRI standardize input

### DISCUSSION THREAD

**Discussion by radu-diaconescu13** [facebookresearch/fastMRI#338] [Q&A] [answered]
Hello,

I have a question about how the data should be normalized for Fast MRI.
From what i have seen in the repo, the following steps are made:

    get mean and std of input image
    normalize it with these
    use the same mean and std to normalized also the target image
    do the inverse operation (multiply by std and add mean) applied on the target image and on the model predictions before computing the metrics

Is my understanding correct? If yes, why are we always using the input image and std? is it because ideally we are dealing with the same image in the end and so we expect to have the same statistics?
Additionally, why is normalization done this way, versus just dividing by the max values, for example? Isnt the batch norm in the network supposed to standardize by batch already?

Finally, why do we need normalization at all if we have batch norm?

Asking a lot of questions, because I am trying to learn :sweat_smile: 

Thank you

**Comment by mmuckley** [comment, marked as answer]
Hello @radu-diaconescu13, I will try to answer your questions.

1. I think your understanding is correct, except I don't think we normalize the target image at all.
2. Dividing by the max value is also acceptable. There are actually a number of ways to normalize, and I've played with quite a few. The method in the repository gives pretty good performance, as well as stable training, which is probably why they settled on it 6 years ago.
3. Image normalization functions differently from layer normalizations. Layer normalizations largely help with optimization, while data normalizations help remove biases and covariances that are well-structured, which helps train the neural network. For example, in image classification models you will see both BatchNorm and mean/standard deviation operations on ImageNet images.

**Reply by radu-diaconescu13** [reply]
Hello @mmuckley 

Thank you for you answers.

1. I saw the below code in fastmri/data/transforms.py in UnetTransform. So you are saying line 376 is not actually needed maybe? Furthermore, is the clamping applied to image and target important? I am asking because we went down the route of rewriting the code in vanilla pytorch, only taking pieces from the fastmri repo as needed.

![image](https://github.com/facebookresearch/fastMRI/assets/32652089/f1f56b70-be7f-49c8-a650-d77051a78738)

2. Thank you.

3. Makes sense.

**Reply by mmuckley** [reply]
Hello @radu-diaconescu13 I think that line probably is necessary because I don't think the U-Net reconstructor unnormalizes before outputting the image.

But for VarNet I think unnormalization is included in the model.

**Reply by radu-diaconescu13** [reply]
I see. And because we use Unet, we have to include it. We will double check the constructor. Thank you again!
